iree/base/status_test.cc

```diff
@@ -23,43 +23,60 @@ namespace {
 using ::iree::testing::status::StatusIs;
 using ::testing::HasSubstr;
 
+#if (IREE_STATUS_FEATURES & IREE_STATUS_FEATURE_ANNOTATIONS) != 0
+#define CHECK_STATUS_MESSAGE_SUBSTR(status, message_substr) \
+  EXPECT_THAT(status.ToString(), HasSubstr(message_substr))
+#else
+#define CHECK_STATUS_MESSAGE_SUBSTR(status, message_substr) \
+  EXPECT_THAT(status.ToString(), HasSubstr(StatusCodeToString(status.code())));
```

Comment:
It seems weird to have an assertion that uses the thing you're asserting on. Maybe better to have the caller pass a code and a message and have the message part just get ignored if messages aren't used

Reply:
It doesn't seem strange to me. Test helpers should be practical and simple.

Reply:
But it means you're just duplicating the code under test. What's the point of this assertion? If someone wants to assert that the message contains the status code then they should just do that... For the !IREE_STATUS_FEATURE_ANNOTATIONS case, this check can just do nothing (or cast things to void to guarantee they're used)

It's especially weird for the `CHECK_STREAM_SUBSTR` case, where it means passing a whole extra status parameter just for its code.

Also isn't the code included in the message even if there is a string message? Why don't we always check for it? I think it would make sense to have the no-annotations case just do strictly less than the annotations case. Otherwise your assertion is asserting on something somewhat unrelated from what it looks like in the test body. I wrote a suggested edit below